### PR TITLE
Cloud component db

### PR DESCRIPTION
**Pull request by napowderly**
Components are now fetched from ato component API!

Screenshot 2024-01-12 at 6 08 13 PM

Pretty simply, we just send all the data we have about the component, and the server finds a part using the specified parameters and returns a JSON with the part data.

### CODE REVIEW

**Review by mawildoer — CHANGES_REQUESTED** [review]
There's a few comments that got orphaned on Gitlab as we moved over to Github.
Would you be able to address those before we merge this?https://gitlab.atopile.io/atopile/atopile/-/merge_requests/127

File: src/atopile/components.py
Comment (mawildoer) [inline src/atopile/components.py:135]: These should probably be at the top of the file - they're currently casuing linting failures

File: src/atopile/front_end.py
Comment (mawildoer) [inline src/atopile/front_end.py:84]: This is typed as a `Pint` unit. It should be converted to a string like this on ingestion. Whatever is using this should do the conversion to a string - or, if this does warrant changing, the typing needs changing too

File: src/atopile/front_end.py
Comment (mawildoer) [inline src/atopile/front_end.py:106]: nit; this should be named `to_dict` since it's returning a dict, not a JSON string.

File: src/atopile/cli/create.py
Comment (mawildoer) [inline src/atopile/cli/create.py:7]: nit; (wouldn't bother except there's two other linting errors blocking the PR)
Run "Organise Imports" command in VSCode, I believe this shouldn't have moved (and can impact merges if someone else has touched imports)

File: src/atopile/components.py
Comment (mawildoer) [inline src/atopile/components.py:59]: I think the comment I left on Gitlab still applies here, it makes the CLI inconsistent. Basically, this means your CWD needs to be inside the project you want to work on and you can't just say `ato build path/to/project/file.ato...`

File: src/atopile/components.py
Comment (mawildoer) [inline src/atopile/components.py:162]: I'd generally recommend using an `attrs` object or [`dataclass`](https://docs.python.org/3/library/dataclasses.html) for things like this. It's really easy to get dict keys wrong and type checkers can't help you with it.

File: src/atopile/front_end.py
Comment (mawildoer) [inline src/atopile/front_end.py:82]: re-adding this comment; I don't think we should be casting this to a string, or, if there's a reason we should be doing so, we need to also change the type-hint.

File: src/atopile/components.py
Comment (napowderly) [reply src/atopile/components.py]: Done

File: src/atopile/components.py
Comment (napowderly) [reply src/atopile/components.py]: Fixed

File: src/atopile/components.py
Comment (napowderly) [reply src/atopile/components.py]: fixed

File: src/atopile/cli/create.py
Comment (napowderly) [reply src/atopile/cli/create.py:7]: fixed

File: src/atopile/components.py
Comment (napowderly) [reply src/atopile/components.py]: Updated

File: src/atopile/components.py
Comment (napowderly) [reply src/atopile/components.py:60]: Done

**Review by mawildoer — APPROVED** [review]

**Pull request merged**
